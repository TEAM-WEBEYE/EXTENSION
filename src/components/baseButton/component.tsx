import React from "react";

interface BaseButtonProps {
    children: React.ReactNode;
    onClick: () => void;
    ariaLabel?: string;
    color?: "gray" | "white" | "purple" | "black" | "yellow" | "dark";
    isSelected?: boolean;
}

const buttonStyles = {
    gray: "bg-grayscale-200 text-grayscale-900 border-grayscale-300 hover:bg-grayscale-200 hover:border-grayscale-300 hover:text-grayscale-400 ",
    white: "bg-grayscale-100 text-grayscale-900 border-grayscale-300 hover:bg-grayscale-100 hover:border-grayscale-300 hover:text-grayscale-500",
    purple: "bg-purple-default text-grayscale-100 border-none hover:bg-purple-light hover:border-none hover:text-grayscale-100",
    yellow: "bg-yellow-default text-grayscale-900 border-none hover:bg-yellow-light hover:border-none hover:text-grayscale-600",
    black: "bg-grayscale-900 text-grayscale-100 border-grayscale-800 hover:bg-grayscale-800 hover:border-grayscale-800 hover:text-grayscale-500",
    dark: "bg-grayscale-800 text-grayscale-100 border-grayscale-800 hover:bg-grayscale-700 hover:border-grayscale-800 hover:text-grayscale-500",
};

const selectedStyles = {
    gray: "bg-grayscale-200 border-purple-default text-grayscale-900",
    white: "bg-grayscale-100 border-purple-default text-grayscale-900",
    purple: "bg-purple-default border-purple-default text-grayscale-100",
    yellow: "bg-yellow-default border-purple-default text-grayscale-900",
    black: "bg-grayscale-900 border-purple-default text-grayscale-100",
    dark: "bg-grayscale-800 border-purple-default text-grayscale-100",
};

const baseClasses =
    "rounded-[14px] w-[140px] h-[140px] flex items-center justify-center border-4 border-solid font-24-Regular font-koddi";

export function BaseButton({
    children,
    onClick,
    ariaLabel,
    color = "gray",
    isSelected = false,
}: BaseButtonProps) {
    const colorClasses = isSelected
        ? selectedStyles[color]
        : buttonStyles[color];

    return (
        <button
            className={`${baseClasses} ${colorClasses}`}
            onClick={onClick}
            aria-label={ariaLabel}
            aria-pressed={isSelected}
        >
            {children}
        </button>
    );
}
