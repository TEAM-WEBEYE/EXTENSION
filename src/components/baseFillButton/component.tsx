import { useTheme } from "@src/contexts/ThemeContext";
import React from "react";

interface BaseFillButtonProps {
    children: React.ReactNode;
    onClick: () => void;
    ariaLabel?: string;
    isDisabled?: boolean;
}

export function BaseFillButton({
    children,
    onClick,
    ariaLabel,
    isDisabled = false,
}: BaseFillButtonProps) {
    const { theme, fontClasses } = useTheme();
    const isDarkMode = theme === "dark";

    return (
        <button
            className={`font-koddi ${fontClasses.fontCommon} py-[16px] px-[30px] relative w-full rounded-[14px]  ${
                isDisabled
                    ? isDarkMode
                        ? "bg-grayscale-400 text-grayscale-100"
                        : " text-grayscale-100  bg-grayscale-600"
                    : isDarkMode
                      ? "bg-purple-light text-grayscale-900 hover:opacity-30 active:bg-purple-dark active:hover:opacity-100"
                      : "text-grayscale-100  bg-purple-default hover:opacity-30 active:bg-purple-dark active:hover:opacity-100"
            }`}
            onClick={onClick}
            aria-label={ariaLabel}
            aria-pressed={isDisabled}
            disabled={isDisabled}
        >
            {children}
        </button>
    );
}
