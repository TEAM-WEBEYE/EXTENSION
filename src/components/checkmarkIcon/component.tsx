import React from "react";
import { useTheme } from "@src/contexts/ThemeContext";

export function CheckmarkIcon(
    props: React.SVGProps<SVGSVGElement>,
): JSX.Element {
    const { theme: themeMode } = useTheme();
    const isDarkMode = themeMode === "dark";

    return (
        <svg
            {...props}
            xmlns="http://www.w3.org/2000/svg"
            width={34}
            height={34}
            viewBox="0 0 34 34"
            fill="none"
        >
            <circle
                cx="17"
                cy="17"
                r="15"
                fill={isDarkMode ? "#323335" : "#fefefe"}
            />
            <path
                d="M14.9 19.7L11.675 16.475C11.4 16.2 11.05 16.0625 10.625 16.0625C10.2 16.0625 9.85 16.2 9.575 16.475C9.3 16.75 9.1625 17.1 9.1625 17.525C9.1625 17.95 9.3 18.3 9.575 18.575L13.85 22.85C14.15 23.15 14.5 23.3 14.9 23.3C15.3 23.3 15.65 23.15 15.95 22.85L24.425 14.375C24.7 14.1 24.8375 13.75 24.8375 13.325C24.8375 12.9 24.7 12.55 24.425 12.275C24.15 12 23.8 11.8625 23.375 11.8625C22.95 11.8625 22.6 12 22.325 12.275L14.9 19.7ZM17 32C14.925 32 12.975 31.606 11.15 30.818C9.325 30.03 7.7375 28.9615 6.3875 27.6125C5.0375 26.2635 3.969 24.676 3.182 22.85C2.395 21.024 2.001 19.074 2 17C1.999 14.926 2.393 12.976 3.182 11.15C3.971 9.324 5.0395 7.7365 6.3875 6.3875C7.7355 5.0385 9.323 3.97 11.15 3.182C12.977 2.394 14.927 2 17 2C19.073 2 21.023 2.394 22.85 3.182C24.677 3.97 26.2645 5.0385 27.6125 6.3875C28.9605 7.7365 30.0295 9.324 30.8195 11.15C31.6095 12.976 32.003 14.926 32 17C31.997 19.074 31.603 21.024 30.818 22.85C30.033 24.676 28.9645 26.2635 27.6125 27.6125C26.2605 28.9615 24.673 30.0305 22.85 30.8195C21.027 31.6085 19.077 32.002 17 32Z"
                fill={isDarkMode ? "#B872FF" : "#8914FF"}
            />
        </svg>
    );
}
