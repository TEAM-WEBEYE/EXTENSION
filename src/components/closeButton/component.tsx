import { useAppTheme } from "@src/contexts/ThemeContext";
import React from "react";

interface CloseButtonProps {
    onClick: () => void;
    ariaLabel?: string;
}

export function CloseButton({ onClick, ariaLabel = "닫기" }: CloseButtonProps) {
    const { theme } = useAppTheme();
    const isDarkMode = theme === "dark";

    const strokeColor = isDarkMode ? "#FEFEFE" : "#121212";

    return (
        <button
            className={` p-[16px] relative rounded-[14px] ${
                isDarkMode
                    ? "bg-grayscale-900  hover:opacity-30 border-4 border-solid border-grayscale-700 active:border-purple-light active:hover:opacity-100"
                    : "bg-grayscale-100 hover:opacity-30 border-4 border-solid border-grayscale-300 active:border-purple-default active:hover:opacity-100"
            }`}
            onClick={onClick}
            aria-label={ariaLabel}
        >
            <svg
                xmlns="http://www.w3.org/2000/svg"
                width="36"
                height="36"
                viewBox="0 0 36 36"
                fill="none"
            >
                <path
                    d="M27 8.99805L9 26.998M27 26.998L9 8.99805"
                    stroke={strokeColor}
                    strokeWidth="3"
                    strokeLinecap="round"
                />
            </svg>
        </button>
    );
}
