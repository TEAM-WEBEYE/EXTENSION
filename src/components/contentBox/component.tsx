import { useAppTheme } from "@src/contexts/ThemeContext";
import React from "react";

interface ContentBoxProps {
    children: React.ReactNode;
    ariaLabel?: string;
}

export function ContentBox({ children, ariaLabel }: ContentBoxProps) {
    const { theme, fontClasses } = useAppTheme();

    const isDarkMode = theme === "dark";
    return (
        <div
            className={`${fontClasses.fontCommon} font-koddi px-[24px] py-[18px] rounded-[14px] flex justify-between gap-[110px]
            ${isDarkMode ? `bg-grayscale-800 text-grayscale-100` : `bg-grayscale-200 text-grayscale-900`}`}
            aria-label={ariaLabel}
        >
            {children}
        </div>
    );
}
