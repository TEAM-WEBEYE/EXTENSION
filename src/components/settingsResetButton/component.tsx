import React from "react";
import { useThemeMode } from "@src/contexts/ThemeContext";

interface SettingsResetButtonProps {
    onClick: () => void;
}

export function SettingsResetButton({ onClick }: SettingsResetButtonProps) {
    const { theme: themeMode } = useThemeMode();

    // Tailwind 클래스를 조건부로 적용하기 위한 함수
    const getThemeClasses = () => {
        if (themeMode === "dark" || themeMode === "yellow") {
            return "bg-grayscale-800 text-grayscale-100 hover:text-grayscale-500 active:bg-grayscale-100 active:text-grayscale-900";
        } else {
            return "bg-grayscale-200 text-grayscale-900 hover:text-grayscale-500 active:bg-grayscale-900 active:text-grayscale-100";
        }
    };

    return (
        <button
            className={`px-6 py-[18px] flex gap-[10px] items-center cursor-pointer rounded-[14px] font-24-Bold font-koddi ${getThemeClasses()}`}
            onClick={onClick}
            data-testid="reset-settings-button"
            aria-label="설정 초기화"
        >
            <svg
                xmlns="http://www.w3.org/2000/svg"
                width="36"
                height="36"
                viewBox="0 0 36 36"
                fill="none"
                className="[&>path]:stroke-current"
            >
                <path
                    d="M7.30673 21.75C8.8337 26.5409 13.207 30 18.3636 30C24.7902 30 30 24.6274 30 18C30 11.3726 24.7902 6 18.3636 6C14.0565 6 10.296 8.41319 8.28401 12M11.8182 13.5H6V7.5"
                    strokeWidth="3"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                />
            </svg>
            <div data-testid="reset-settings-text">설정 초기화</div>
        </button>
    );
}
