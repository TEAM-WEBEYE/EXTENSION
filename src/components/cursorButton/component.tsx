import React from "react";
import { BaseButton } from "../baseButton/component";
import { useAppTheme } from "@src/contexts/ThemeContext";
import { getExtensionUrl } from "@src/utils/getExtensionUrl";

interface CursorButtonProps {
    onClick: () => void;
    color?: "white" | "yellow" | "purple";
    size?: "small" | "medium" | "large";
}

export function CursorButton({
    onClick,
    color = "white",
    size = "medium",
}: CursorButtonProps) {
    const { theme } = useAppTheme();

    const adjustedTheme = theme === "light" ? "white" : theme;

    const fileName = `cursor-${size}-${color}.png`;

    return (
        <BaseButton
            onClick={onClick}
            color={adjustedTheme}
            aria-label={`커서 변경 버튼: ${size}, ${color}`}
        >
            <img
                src={getExtensionUrl(fileName)}
                alt={`커서: ${size}, ${color}`}
                className="group-hover:opacity-20"
            />
        </BaseButton>
    );
}
