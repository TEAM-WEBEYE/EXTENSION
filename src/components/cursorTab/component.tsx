import { useTheme } from "@src/contexts/ThemeContext";
import React from "react";
import { CursorButton } from "../cursorButton";

type CursorSize = "small" | "medium" | "large";
type CursorColor = "white" | "black" | "yellow" | "purple" | "mint" | "pink";

const CURSOR_SIZES: CursorSize[] = ["small", "medium", "large"];
const CURSOR_COLORS: CursorColor[] = [
    "white",
    "black",
    "yellow",
    "purple",
    "mint",
    "pink",
];

export function CursorTab() {
    const { theme, fontClasses } = useTheme();
    const { setCursorSize, setCursorTheme, cursorSize, cursorTheme } =
        useTheme();

    const isDarkMode = theme === "dark";

    return (
        <div
            className={`font-koddi p-[18px] rounded-[20px] shadow-[0px_0px_4px_0px_rgba(0,0,0,0.25)] ${
                isDarkMode
                    ? `bg-grayscale-900 text-grayscale-100`
                    : `bg-grayscale-100 text-grayscale-900`
            }`}
        >
            <section className="mb-[26px]">
                <h2 className={fontClasses.fontCommon}>커서 크기 바꾸기</h2>
                <div className="flex gap-5 p-2">
                    {CURSOR_SIZES.map((size) => (
                        <CursorButton
                            key={size}
                            onClick={() => setCursorSize(size)}
                            size={size}
                            isSelected={cursorSize === size}
                        />
                    ))}
                </div>
            </section>

            <section>
                <h2 className={fontClasses.fontCommon}>커서 색상 바꾸기</h2>
                <div className="grid grid-cols-3 gap-5 p-2">
                    {CURSOR_COLORS.map((color) => (
                        <CursorButton
                            key={color}
                            onClick={() => setCursorTheme(color)}
                            color={color}
                            isSelected={cursorTheme === color}
                        />
                    ))}
                </div>
            </section>
        </div>
    );
}
