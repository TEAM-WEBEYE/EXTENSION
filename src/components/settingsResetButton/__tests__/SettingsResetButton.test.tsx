import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { SettingsResetButton } from "../component";
import { ThemeProvider, useThemeMode } from "@src/contexts/ThemeContext";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

const mockLocalStorage = (() => {
    let store: Record<string, string> = {};
    return {
        getItem: jest.fn((key: string) => store[key] || null),
        setItem: jest.fn((key: string, value: string) => {
            store[key] = value;
        }),
        clear: jest.fn(() => {
            store = {};
        }),
    };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });

jest.mock("@src/contexts/ThemeContext", () => {
    const originalModule = jest.requireActual("@src/contexts/ThemeContext");
    return {
        ...originalModule,
        useThemeMode: jest.fn(),
    };
});

describe("SettingsResetButton", () => {
    const mockOnClick = jest.fn();

    beforeEach(() => {
        mockLocalStorage.clear();
        jest.clearAllMocks();

        (useThemeMode as jest.Mock).mockImplementation(() => ({
            theme: "light",
            setTheme: jest.fn(),
        }));
    });

    const themeModes = ["light", "dark", "yellow"];

    themeModes.forEach((themeMode) => {
        it(`${themeMode} 테마에서 스냅샷이 일치하는지 확인`, () => {
            mockLocalStorage.getItem.mockReturnValueOnce(themeMode);

            (useThemeMode as jest.Mock).mockImplementation(() => ({
                theme: themeMode,
                setTheme: jest.fn(),
            }));

            const tree = renderer
                .create(
                    <ThemeProvider>
                        <SettingsResetButton onClick={jest.fn()} />
                    </ThemeProvider>,
                )
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it(`${themeMode} 테마에서 올바르게 렌더링된다`, () => {
            mockLocalStorage.getItem.mockReturnValueOnce(themeMode);

            (useThemeMode as jest.Mock).mockImplementation(() => ({
                theme: themeMode,
                setTheme: jest.fn(),
            }));

            render(
                <ThemeProvider>
                    <SettingsResetButton onClick={mockOnClick} />
                </ThemeProvider>,
            );

            const button = screen.getByTestId("reset-settings-button");
            expect(button).toBeInTheDocument();

            const buttonText = screen.getByTestId("reset-settings-text");
            expect(buttonText).toHaveTextContent("설정 초기화");

            const svgElement = button.querySelector("svg");
            expect(svgElement).toBeInTheDocument();
        });

        it(`${themeMode} 테마에서 버튼 클릭 시 onClick이 호출된다`, () => {
            mockLocalStorage.getItem.mockReturnValueOnce(themeMode);

            (useThemeMode as jest.Mock).mockImplementation(() => ({
                theme: themeMode,
                setTheme: jest.fn(),
            }));

            render(
                <ThemeProvider>
                    <SettingsResetButton onClick={mockOnClick} />
                </ThemeProvider>,
            );

            const button = screen.getByTestId("reset-settings-button");
            fireEvent.click(button);

            expect(mockOnClick).toHaveBeenCalledTimes(1);
        });
    });

    it("올바른 접근성 속성을 가지고 있다", () => {
        render(
            <ThemeProvider>
                <SettingsResetButton onClick={mockOnClick} />
            </ThemeProvider>,
        );

        const button = screen.getByTestId("reset-settings-button");

        expect(button).toHaveAttribute("aria-label", "설정 초기화");
    });

    it("올바른 스타일링 클래스를 가지고 있다", () => {
        render(
            <ThemeProvider>
                <SettingsResetButton onClick={mockOnClick} />
            </ThemeProvider>,
        );

        const button = screen.getByTestId("reset-settings-button");
        expect(button).toHaveClass("px-6");
        expect(button).toHaveClass("py-[18px]");
        expect(button).toHaveClass("flex");
        expect(button).toHaveClass("gap-[10px]");
        expect(button).toHaveClass("items-center");
        expect(button).toHaveClass("cursor-pointer");
        expect(button).toHaveClass("rounded-[14px]");
        expect(button).toHaveClass("font-24-Bold");
        expect(button).toHaveClass("font-koddi");
    });

    it("유효하지 않은 테마인 경우 기본값(light)을 사용한다", () => {
        (useThemeMode as jest.Mock).mockImplementation(() => ({
            theme: "invalid-theme" as any,
            setTheme: jest.fn(),
        }));

        render(
            <ThemeProvider>
                <SettingsResetButton onClick={mockOnClick} />
            </ThemeProvider>,
        );

        const button = screen.getByTestId("reset-settings-button");
        expect(button).toBeInTheDocument();
        expect(screen.getByTestId("reset-settings-text")).toHaveTextContent(
            "설정 초기화",
        );
    });
});
