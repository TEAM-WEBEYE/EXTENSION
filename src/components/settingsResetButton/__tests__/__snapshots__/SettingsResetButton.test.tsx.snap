// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsResetButton dark 테마에서 스냅샷이 일치하는지 확인 1`] = `
<button
  aria-label="설정 초기화"
  className="px-6 py-[18px] flex gap-[10px] items-center cursor-pointer rounded-[14px] font-24-Bold font-koddi css-ixicwg"
  data-testid="reset-settings-button"
  onClick={[MockFunction]}
>
  <svg
    fill="none"
    height="36"
    viewBox="0 0 36 36"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.30673 21.75C8.8337 26.5409 13.207 30 18.3636 30C24.7902 30 30 24.6274 30 18C30 11.3726 24.7902 6 18.3636 6C14.0565 6 10.296 8.41319 8.28401 12M11.8182 13.5H6V7.5"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="3"
    />
  </svg>
  <div
    data-testid="reset-settings-text"
  >
    설정 초기화
  </div>
</button>
`;

exports[`SettingsResetButton light 테마에서 스냅샷이 일치하는지 확인 1`] = `
<button
  aria-label="설정 초기화"
  className="px-6 py-[18px] flex gap-[10px] items-center cursor-pointer rounded-[14px] font-24-Bold font-koddi css-ehbcqu"
  data-testid="reset-settings-button"
  onClick={[MockFunction]}
>
  <svg
    fill="none"
    height="36"
    viewBox="0 0 36 36"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.30673 21.75C8.8337 26.5409 13.207 30 18.3636 30C24.7902 30 30 24.6274 30 18C30 11.3726 24.7902 6 18.3636 6C14.0565 6 10.296 8.41319 8.28401 12M11.8182 13.5H6V7.5"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="3"
    />
  </svg>
  <div
    data-testid="reset-settings-text"
  >
    설정 초기화
  </div>
</button>
`;

exports[`SettingsResetButton yellow 테마에서 스냅샷이 일치하는지 확인 1`] = `
<button
  aria-label="설정 초기화"
  className="px-6 py-[18px] flex gap-[10px] items-center cursor-pointer rounded-[14px] font-24-Bold font-koddi css-ixicwg"
  data-testid="reset-settings-button"
  onClick={[MockFunction]}
>
  <svg
    fill="none"
    height="36"
    viewBox="0 0 36 36"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.30673 21.75C8.8337 26.5409 13.207 30 18.3636 30C24.7902 30 30 24.6274 30 18C30 11.3726 24.7902 6 18.3636 6C14.0565 6 10.296 8.41319 8.28401 12M11.8182 13.5H6V7.5"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="3"
    />
  </svg>
  <div
    data-testid="reset-settings-text"
  >
    설정 초기화
  </div>
</button>
`;
